src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,81 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.HashMap;
+
 public class WorkWithFile {
+    private static final String SUPPLY_KEY = "supply";
+    private static final String BUY_KEY = "buy";
+    private static final String RESULT_KEY = "result";
+    private static final String LINE_SPLIT_KEY = "\n";
+    private static final String CSV_SPLIT_KEY = ",";
+    private static final int OPERATION_TYPE_INDEX = 0;
+    private static final int VALUE_INDEX = 1;
+
     public void getStatistic(String fromFileName, String toFileName) {
+        HashMap<String, Integer> operationTotals = processData(fromFileName);
+        String report = generateReport(operationTotals);
+        writeToFile(toFileName, report);
+    }
+
+    private String read(String fromFileName) {
+        StringBuilder stringBuilder = new StringBuilder();
+        try (BufferedReader bufferedReader = new BufferedReader(new FileReader(fromFileName))) {
+            String line;
+            while ((line = bufferedReader.readLine()) != null) {
+                stringBuilder.append(line).append(LINE_SPLIT_KEY);
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("Can't read file! " + fromFileName, e);
+        }
+        return stringBuilder.toString();
+    }
+
+    private HashMap<String, Integer> processData(String fromFileName) {
+        HashMap<String, Integer> operationTotals = new HashMap<>();
+        String fileContent = read(fromFileName);
```

Comment:
would be nice to split it even more and call `read()` outside of this method, so that `processData()` would receive only `fileContent` and worked with that only